packages/at_secondary_server/test/sync_unit_test.dart

```diff
@@ -975,124 +922,157 @@ void main() {
         expect(syncResponseList[0]['operation'], '*');
       });
-      tearDown(() async => await tearDownMethod());
+      tearDown(() async => await verbTestsTearDown());
     });
 
     group('A group of tests on APKAM Enrollment', () {
-      setUp(() async => await setUpMethod());
-      test(
-          'A test to verify keys whose namespace are enrolled are only returned',
-          () async {
-        await secondaryPersistenceStore!
-            .getSecondaryKeyStore()
-            ?.put('public:phone.wavi@alice', AtData()..data = '8897896765');
-        await secondaryPersistenceStore!
-            .getSecondaryKeyStore()
-            ?.put('public:mobile.buzz@alice', AtData()..data = '8897896765');
-        var enrollmentId = Uuid().v4();
-        String enrollmentKey =
-            '$enrollmentId.$newEnrollmentKeyPattern.$enrollManageNamespace@alice';
-        final enrollJson = {
-          'sessionId': '123',
-          'appName': 'wavi',
-          'deviceName': 'pixel',
-          'namespaces': {'wavi': 'rw'},
-          'apkamPublicKey': 'testPublicKeyValue',
-          'requestType': 'newEnrollment',
-          'approval': {'state': 'approved'}
-        };
-        await secondaryPersistenceStore!
-            .getSecondaryKeyStore()
-            ?.put(enrollmentKey, AtData()..data = jsonEncode(enrollJson));
+      late SyncProgressiveVerbHandler handler;
+      setUp(() async {
+        await verbTestsSetUp();
+        handler = SyncProgressiveVerbHandler(secondaryKeyStore);
+      });
 
-        var syncProgressiveVerbHandler = SyncProgressiveVerbHandler(
-            secondaryPersistenceStore!.getSecondaryKeyStore()!);
+      Future<List> executeSyncVerb(
+        String enrollmentId, {
+        String fromCommitId = '-1',
+        String? skipDeletesUntil,
+      }) async {
         var response = Response();
-        var inBoundSessionId = '_6665436c-29ff-481b-8dc6-129e89199718';
-        var atConnection = InboundConnectionImpl(mockSocket, inBoundSessionId);
-        atConnection.metaData.isAuthenticated = true;
-        (atConnection.metaData as InboundConnectionMetadata).enrollmentId =
-            enrollmentId;
+        inboundConnection.metaData.isAuthenticated = true;
+        inboundConnection.metaData.enrollmentId = enrollmentId;
         var syncVerbParams = HashMap<String, String>();
-        syncVerbParams.putIfAbsent(AtConstants.fromCommitSequence, () => '-1');
-        await syncProgressiveVerbHandler.processVerb(
-            response, syncVerbParams, atConnection);
-        List syncResponseList = jsonDecode(response.data!);
+        syncVerbParams[AtConstants.fromCommitSequence] = fromCommitId;
+        if (skipDeletesUntil != null) {
+          syncVerbParams[AtConstants.skipDeletesUntil] = skipDeletesUntil;
+        }
+        await handler.processVerb(response, syncVerbParams, inboundConnection);
+        return jsonDecode(response.data!);
+      }
```

Comment:
New test: sync with sub-namespace permissions. Extended and renamed the existing test (namespace permissions)